src/datadog_directive.cpp

```diff
@@ -1028,5 +1032,17 @@ char *hijack_auth_request(ngx_conf_t *cf, ngx_command_t *command,
   return static_cast<char *>(NGX_CONF_ERROR);
 }
 
+char *waf_thread_pool_name(ngx_conf_t *cf, ngx_command_t *command,
+                           void *conf) noexcept {
+  datadog_loc_conf_t *loc_conf = static_cast<datadog_loc_conf_t *>(conf);
```

Comment:
We do not create thread pools explicitly. The user needs to create them. The reason for this is that there are important settings (the number of threads in the pool and the queue depth) that are better left to the user, because they interact with other user-controlled settings like the max number of workers. The configuration is like this:

```conf
thread_pool waf_thread_pool threads=2 max_queue=5;

http {
    datadog_waf_thread_pool_name waf_thread_pool;
}
```

you can create several pools and then use different ones from different locations, though I haven't tested that yet.

Reply:
[NGINX Development guide](https://nginx.org/en/docs/dev/development_guide.html#threads) states:
> [...] calling ngx_thread_pool_add(cf, name), which either creates a new thread pool with the given name or returns a reference to the pool with that name if it already exists.

This approach can still inadvertently create a thread pool by directive per location. While the rationale behind is clear, there's a concern regarde its usage by customers who might not fully understand its implications, like, lots of "empty" thread pool (a large chunk of users copy/paste config). 

One way to address this concern is by creating a thread pool for WAF with an hardcoded name, such as `waf_thread_pool`. Savvy users , will still have the flexibility to configure it using `thread_pool waf_thread_pool threads=X max_queue=Y;`

That's just a suggestion.

Reply:
I've changed it to `ngx_thread_pool_get`. This way the thread is not implicitly created and the user _must_ specify the thread pool to use (o/wise appsec will be disabled for that location).

It would arguably be more user friendly (though questionable for production purposes) to automatically create a thread pool, but we can leave such niceties for later.